backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/rest/WsProjectService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Verifa Oy, 2018. Part of the SW360 Project.
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.sw360.wsimport.rest;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import org.apache.log4j.Logger;
+import org.eclipse.sw360.datahandler.thrift.projectimport.TokenCredentials;
+import org.eclipse.sw360.wsimport.domain.*;
+import org.eclipse.sw360.wsimport.utility.WsTokenType;
+
+import java.io.IOException;
+
+/**
+ * @author ksoranko@verifa.io
+ */
+public class WsProjectService {
+
+    private static final Logger LOGGER = Logger.getLogger(WsProjectService.class);
+    private static final WsRestClient restClient = new WsRestClient();
+    private static final Gson gson = new Gson();
+
+    public WsProject getWsProject(String projectToken, TokenCredentials tokenCredentials) throws JsonSyntaxException {
+        String projectVitalString = null;
+        try {
+            projectVitalString = restClient.getData("getProjectVitals", projectToken, WsTokenType.PROJECT, tokenCredentials);
+        } catch (IOException ioe) {
+            LOGGER.error(ioe);
+        }
+        WsProjectVitals wsProjectVitals = gson.fromJson(projectVitalString, WsProjectVitals.class);
+        WsProjectVitalInformation projectVitalInformation = wsProjectVitals.getProjectVitals()[0];
```

Comment:
In case of an exception (IOException) you will raise NPE with `getProjectVitals()[0]` and

gson.fromJson raises NPE if input is null
```

  public JsonReader(Reader in) {
    if (in == null) {
      throw new NullPointerException("in == null");
    }
    this.in = in;
  }
```

Reply:
In case of an IOE your `projectVitalString` in method `restClient.getData` is null
`String projectVitalString = null;`
You just log this error `LOGGER.error(ioe);`
After this you deserialilze the null value of `projectVitalString` with gson.
Gson will throw a NPE. You should check if `projectVitalString` is not empty and valid.

Reply:
You should also check if `wsProjectVitals.getProjectVitals` exists before ` wsProjectVitals.getProjectVitals[0]` (get element from it)